### PR TITLE
auto save timesheet entry

### LINKED ISSUE #1495: auto saving of timesheet entry to local storage until it is saved
### Is your proposal related to a problem?

auto saving of timesheet entry to local storage until it is saved. 

### Describe the solution you'd like

1. Login as any user 
2. On the time tracking tab, click on new entry 
3. Fill the required fields and refresh the screen 
4. After page refresh the previously filled details should be loaded on the page 

### Requirement explanation video
https://www.loom.com/share/bb2907fec1264066ba38299aa46c9a39

### PR DESCRIPTION
**Pull request by apoorv1316**
Closes #1495

### CODE REVIEW

**Comment by supriya3105** [comment]
@prasanthchaduvula Please review the PR.

File: app/javascript/src/components/TimeTracking/EntryForm.tsx [hunk 1]
```diff
@@ -178,6 +202,17 @@ const AddEntry: React.FC<Iprops> = ({
     handleFillData();
   }, []);
 
+  useEffect(() => {
+    // Save the entry data to localStorage whenever it changes
+    localStorage.setItem("note", note);
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/TimeTracking/EntryForm.tsx:207]: Move this to a function and call it inside the useEffect

File: app/javascript/src/components/TimeTracking/EntryForm.tsx [hunk 2]
```diff
@@ -37,13 +37,30 @@ const AddEntry: React.FC<Iprops> = ({
...
+  const [project, setProject] = useState<string>(
+    localStorage.getItem("project") || ""
+  );
+
+  const [projectId, setProjectId] = useState<number>(
+    parseInt(localStorage.getItem("projectId") || "0")
+  );
+
+  const [billable, setBillable] = useState<boolean>(
+    localStorage.getItem("billable") === "true"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/TimeTracking/EntryForm.tsx:58]: Move the setting of the state values to a method and call them inside the useEffect and use `setToLocalStorage` and `getValueFromLocalStorage` method from `utils/storage`

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/TimeTracking/EntryForm.tsx (hunk 1 above)
Comment (apoorv1316) [reply app/javascript/src/components/TimeTracking/EntryForm.tsx]: Done

File: app/javascript/src/components/TimeTracking/EntryForm.tsx (hunk 2 above)
Comment (apoorv1316) [reply app/javascript/src/components/TimeTracking/EntryForm.tsx]: Done

**Comment by prasanthchaduvula** [comment]
@apoorv1316 I have tested this PR. When we click on New time entry and add the description. After that If we refresh or visit clients and come back to a time-tracking page and click on a new timesheet entry the previous details are not preloaded

**Comment by apoorv1316** [comment]
@prasanthchaduvula It is working. Please check again.

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**